openedx/adg/admin_dashboard/views.py

```diff
@@ -0,0 +1,55 @@
+"""
+Contains all the views related code that'll be used for the admin_dashboard package and any apps it contains
+"""
+from django.contrib.auth.mixins import AccessMixin
+from django.contrib.auth.views import redirect_to_login
+from django.http import HttpResponse
+from django.views.generic import TemplateView
+from rest_framework.status import HTTP_400_BAD_REQUEST
+
+from openedx.adg.lms.applications.helpers import has_admin_permissions
+
+
+class LoginRequiredAndUserTestPassesMixin(AccessMixin):
+    """
+    User must be authenticated and must pass the pre-condition defined by `is_precondition_satisfied`
+    in order to access the view using this mixin
+    """
+
+    def dispatch(self, request, *args, **kwargs):
+        """
+        Redirects to login page for unauthenticated users. Runs `handle_no_permission` if preconditions are not
+        satisfied. Runs the view normally for authenticated user with proper conditions satisfied.
+        """
+        if not request.user.is_authenticated:
+            return redirect_to_login(request.get_full_path(), self.get_login_url(), self.get_redirect_field_name())
+        elif not self.is_precondition_satisfied():
+            return self.handle_no_permission()
+        return super().dispatch(request, *args, **kwargs)
+
+
+class EnsureAdminUserAuthenticatedMixin(LoginRequiredAndUserTestPassesMixin):
+    """
+    Mixin that makes sure that the user is authenticated and is an ADG admin, BU admin or Superuser.
+    If the user is unauthenticated, it redirects them to the login page. If the user is authenticated
+    but not an ADG admin or BU Admin or Superuser, they are given the HTTP 400 error response
+    """
+
+    def is_precondition_satisfied(self):
+        """
+        Checks if a user is an ADG admin or BU Admin or Superuser. Failure to be any type of admin
+        fails the condition and user is given the 400 error
+        """
+        return has_admin_permissions(self.request.user)
+
+    def handle_no_permission(self):
+        """
+        Shows the HTTP 400 error response to the user.
+        """
+        return HttpResponse(status=HTTP_400_BAD_REQUEST)
+
+
+class RestrictedTemplateView(EnsureAdminUserAuthenticatedMixin, TemplateView):
+    """
+    TemplateView that is only accessible to ADG admins, BU Admins and superusers
+    """
```

Comment:
let's not leave the class empty and instead add a `pass` statement.

Reply:
Pylint gave me an error on doing so, that is why I removed it.

Reply:
What was the error? Maybe we can suppress it, because throughout the open edx platform, empty classes are declared with the `pass` statement; and that is the [correct way to declare an empty class as well](https://docs.python.org/3/tutorial/classes.html#odds-and-ends).

Reply:
I realize that. The error was 'W0107: Unnecessary pass statement (unnecessary-pass)'

Reply:
We can suppress the pylint warning and add the pass statement anyway.